python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1001 @@
+        if hasattr(value, "array"):
+            self._array[filterIndex, sy, sx] = value.array
+        else:
+            self._array[filterIndex, sy, sx] = value
+
+
+def imageFromSingles(cls, filters, singles):
+    """Construct a MultibandImage from a collection of single band images
+
+    Parameters
+    ----------
+    filters: list
+        List of filter names.
+    singles: list
+        A list of single band objects.
+        If `array` is not `None`, then `singles` is ignored
+    """
+    array = np.array([image.array for image in singles], dtype=singles[0].array.dtype)
+    if not np.all([image.getBBox() == singles[0].getBBox() for image in singles[1:]]):
+        raise ValueError("Single band images did not all have the same bounding box")
+    bbox = singles[0].getBBox()
+    return cls(filters, array, bbox)
+
+
+def imageFactory(cls, filters, filterKwargs, singleType=ImageF, **kwargs):
+    """Build a MultibandImage from a set of keyword arguments
+
+    Parameters
+    ----------
+    filters: list
+        List of filter names.
+    singleType: class
+        Class of the single band objects.
+        This is ignored unless `singles` and `array`
+        are both `None`, in which case it is required.
+    filterKwargs: dict
+        Keyword arguments to pass to `_fullInitialize` to
+        initialize a new instance of an inherited class
+        that are different for each filter.
+        The keys are the names of the arguments and the values
+        should also be dictionaries, with filter names as keys
+        and the value of the argument for a given filter as values.
+    kwargs: dict
+        Keyword arguments to pass to `_fullInitialize` to
+        initialize a new instance of an inherited class that are the
+        same in all bands.
+    """
+    # Attempt to load a set of images
+    singles = []
+    for f in filters:
+        if filterKwargs is not None:
+            for key, value in filterKwargs:
+                kwargs[key] = value[f]
+        singles.append(singleType(**kwargs))
+    return cls.imageFromSingles(filters, singles)
+
+
+class MultibandImage(MultibandImageBase):
+    """Multiband Image class
+
+    See `MultibandImageBase` for a description of the parameters.
+    """
+    def __init__(self, filters, array, bbox=None):
+        super().__init__(filters, array, Image, bbox)
+
+    @staticmethod
+    def fromImages(filters, singles):
+        """Construct a MultibandImage from a collection of single band images
+
+        see `fromSingles` for a description of parameters
+        """
+        return imageFromSingles(MultibandImage, filters, singles)
+
+    @staticmethod
+    def Factory(filters, filterKwargs, singleType=ImageF, **kwargs):
+        """Build a MultibandImage from a set of keyword arguments
+
+        see `imageFactory` for a description of parameters
+        """
+        return imageFactory(MultibandImage, filters, filterKwargs, singleType, **kwargs)
```

Comment:
The main reason I wrote this was to give users the ability to pass a butler and create a `MultibandExposure` from a collection of keyword arguments, some of which may differ by filter. If we think that it's too much of a burden to support that I can remove both `imageFactory` and `tripleFactory`, otherwise I can just rename them something like `makeImageFromKwargs`.

Reply:
I'm not opposed to keeping anything anything you have a use case for.